src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -570,6 +570,22 @@ public static void CheckCanGenerateEEType(NodeFactory factory, TypeDesc type)
                     // TODO: validate constraints
                 }
 
+                // Check the type doesn't have bogus MethodImpls or overrides and we can get the finalizer.
+                defType.GetFinalizer();
+
+                const int FieldOffsetMax = (1 << 27) - 1; // FIELD_OFFSET_MAX in CoreCLR
+                const int FieldOffsetLastRealOffset = FieldOffsetMax - 6; // FIELD_OFFSET_LAST_REAL_OFFSET in CoreCLR
```

Comment:
I tend to look at this the same as the `32` limit on array ranks. ECMA doesn't specify a limit and there's nothing wrong with someone creating a 60-dim array with all zero lengths. 

The purpose of the limit is to force user code to be portable across our runtimes...